samples/bluetooth/central_uart/src/main.c

```diff
@@ -486,32 +478,88 @@ static int nus_client_init(void)
 BT_SCAN_CB_INIT(scan_cb, scan_filter_match, NULL,
 		scan_connecting_error, scan_connecting);
 
-static int scan_init(void)
+static void try_add_address_filter(const struct bt_bond_info *info, void *user_data)
 {
 	int err;
-	struct bt_scan_init_param scan_init = {
-		.connect_if_match = 1,
-	};
+	char addr[BT_ADDR_LE_STR_LEN];
 
-	bt_scan_init(&scan_init);
-	bt_scan_cb_register(&scan_cb);
+	bt_addr_le_to_str(&info->addr, addr, sizeof(addr));
+
+	struct bt_conn *conn = bt_conn_lookup_addr_le(BT_ID_DEFAULT, &info->addr);
+	struct bt_conn_info conn_info;
```

Comment:
variables should be declared at top of scope

Reply:
The variable is not used, it has been removed.